src/Tizen.NUI.WindowSystem/src/internal/Interop/Interop.InputGesture.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Tizen.NUI.WindowSystem
+{
+    internal static partial class Interop
+    {
+        internal static class InputGesture
+        {
+            const string lib = "libcapi-ui-efl-util.so.0";
+
+            internal static string LogTag = "Tizen.NUI.WindowSystem";
+
+            [DllImport(lib, EntryPoint = "efl_util_gesture_initialize")]
+            internal static extern SafeGestureHandle Initialize();
+
+            [DllImport(lib, EntryPoint = "efl_util_gesture_deinitialize")]
+            internal static extern ErrorCode Deinitialize(SafeGestureHandle gestureHandle);
```

Comment:
I check you'd change to use `SafeHandle` instead of NUI dispose pattern.

But `SafeHandle.Dispose(false)` could be invoke at worker thread(Actually, GC thread)

Is all Deinitialize API's what you where made are thread safe?

Reply:
@hinohie , thank you for your comment.
Do you recommend to use NUI dispose pattern instead of SafeHandle?
I have modified my patch to use SafeHandle according to the previous comment (following).
---------------
@[myroot](https://github.com/myroot) myroot [last week](https://github.com/Samsung/TizenFX/pull/6184#discussion_r1654572655)
Use C# standard Dispose pattern instead NUI Dispose pattern

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern

Reply:
could you please check the following patch. Mr. Hong seems to be concerned about a situation where dispose does not work in the main thread when disposing is set as false. Therefore, I would like you to refer to the code implemented in NUI2 below and modify it so that we can avoid the previous issue (the problem of being created or disposed on the worker thread) by posting it to the main thread.

https://github.sec.samsung.net/dotnet/nui2/blob/main/src/Tizen.NUI2/NObject.cs#L150

Reply:
@dongsug-song , Is there any reference code snippet in TizenFX? That would be very helpful.

Reply:
@dongsug-song , NUI Dispose pattern으로 다시 변경하는 것이 좋을까요?

Reply:
꼭 필요한 경우를 제외하고는 NUI Dependency 를 끊는 것을 권장하는 추세입니다.

현재 TizenFX 에서 NUI2 와 같은 방식으로 SafeHandle 을 소멸시킨 사용례가 보이지 않아서 불편하시겠지만, NUI2 처럼 Application.PostUI 라는 이벤트를 쓰거나, UI Thread 임이 보장된 상태에서 별도의 eventfd 를 생성 (현 NUI DisposeQueue 의 동작) 하는 등의 방식으로 처리해야 할 것 같습니다 ㅠㅠ

Reply:
@oduna00 
자신 모듈의 thread safe관련 동작은 알고 계셔야 합니다.

제가 확인했던 바로는 SafeHandle로 매핑하라고 했던 저 동작은 단순히 자료구조 메모리 할당 / 해제였습니다.
이 경우에는 TizenFX의 경우 SafeHandle을 사용하도록 가이드 됩니다.
대표적인 예들로 Application F/W의 클래스들이 보통 이 경우에 포함됩니다
https://github.com/Samsung/TizenFX/blob/master/src/Tizen.Applications.Common/Tizen.Applications/AppControl.cs

저희가 SafeHandle로 매핑을 못하고 NUI사례 처럼 IntPtr을 직접 관리하는 경우는 Main thread가 아닌 다른 thread에서 접근 시 아에 동작하지 않는 경우에(thread local 등으로 다른 쓰레드에서 접근이 불가능하는 등..) 한해서, IntPtr과 Finalizer에서 특별한 처리를 거쳐 main thread에서 소멸 되도록 가이드하고 있습니다.

위 경우는 그런 경우가 아니라서 SafeHandle과 일반적인 C# 표준 dispose패턴을 사용하도록 가이드를 하였습니다.

Reply:
@myroot , 자세한 답변 감사드립니다.
SafeHandle을 수정하여 다시 리뷰 올리겠습니다.